meson.build

```diff
@@ -192,6 +192,15 @@ if cc.compiles(malloc_attribute_test, name : 'malloc attribute with arguments')
   add_project_arguments('-DHAVE_MALLOC_EXTENDED_ATTRIBUTE', language: 'c')
 endif
 
+if cc.has_function('closefrom', prefix: '#define _GNU_SOURCE\n#include <unistd.h>')
+  add_project_arguments('-DHAVE_CLOSEFROM', language: 'c')
+endif
+if os == 'Linux' and \
+    cc.has_header_symbol('linux/close_range.h', 'CLOSE_RANGE_CLOEXEC') and \
```

Comment:
use the unistd.h API only.  that's from the C library, not the kernel UAPI.

Reply:
I don't understand this comment. Read the `close_range(2)` man page. The header to include to get the definition of `CLOSE_RANGE_CLOEXEC` is `<linux/close_range.h>`.

Reply:
It's available in `unistd.h` as well, mimicking e.g. FreeBSD. It looks like man-pages (the project) needs to be updated.

Reply:
C library APIs provide all the defines/values you need.  if the C library provides `close_range()`, then it provides the various flags/bits too.

```sh
$ gcc -E -dD -D_GNU_SOURCE - <<<'#include <unistd.h>' | grep CLOSE_RANGE_CLOEXEC
#define CLOSE_RANGE_CLOEXEC (1U << 2)
$ qlist sys-libs/glibc | xargs grep CLOSE_RANGE_CLOEXEC
/usr/include/bits/unistd_ext.h:#ifndef CLOSE_RANGE_CLOEXEC
/usr/include/bits/unistd_ext.h:# define CLOSE_RANGE_CLOEXEC (1U << 2)
```

Reply:
I think these APIs hadn't yet made it into Glibc when I opened this PR. I've amended it now. Thanks.

Reply:
fwiw, afaict, the APIs were in glibc-2.34 released in Aug 2021.

it does look like the close_range(2) man page is outdated as Sam noted, and closefrom(2) is missing entirely.  Basically only docs for the kernel interfaces have been written ... none for the C library yet.

Reply:
I've filed https://bugzilla.kernel.org/show_bug.cgi?id=216951 as well.